src/loader/bgw_launcher.c

```diff
@@ -610,6 +613,32 @@ ts_bgw_cluster_launcher_main(PG_FUNCTION_ARGS)
 	PG_RETURN_VOID();
 }
 
+/*
+ * Inside the entrypoint, we must check again if we're in a template db
+ * even though we excluded template dbs in populate_database_htab because
+ * we can be called on, say, CREATE EXTENSION in a template db and then
+ * we'll not stop til next server shutdown so if we hit this point and are
+ * in a template db, we throw an error and shut down Check in the syscache
```

Comment:
Ugh sorry super duper minor nit. Period in between "down" and "Check"? Otherwise LGTM

Reply:
hmmm...I think that was actually pgindent. That was on a separate line before. Odd. I can fix at some future point and will look out for that in pgindent in future.